src/pybind/mgr/volumes/fs/subvolume.py

```diff
@@ -141,6 +141,34 @@ def remove_subvolume(self, spec, force):
         except cephfs.Error as e:
             raise VolumeException(-e.args[0], e.args[1])
 
+    def resize_subvolume(self, spec, newsize=None, extend=True):
+        """
+        :param spec: subvolume path specification
+        :param newsize: new size In bytes, or None for no size limit
+        :param extend: the extend/shrink switch
+        :return: None
+        """
+
+        if newsize == None or newsize <= 0:
+            raise VolumeException(-errno.EINVAL, "Provide a valid size")
+
+        subvolpath = spec.subvolume_path #check if path is valid?
+        maxbytes = int(self.fs.getxattr(subvolpath, 'ceph.quota.max_bytes').decode('utf-8'))
+
+        if extend:
+            if newsize <= maxbytes:
+                raise VolumeException(-errno.EINVAL, "Can't extend the subvolume. "
+                                      "The new size must be greater than the current size '{0}'".format(maxbytes))
+        else:
+            if newsize >= maxbytes:
```

Comment:
Should we handle the case where used subvolume size is greater than the size to be shrunk to, as done in https://github.com/openstack/manila/commit/9ff0d1312#diff-4256e8eb2400d0325fe689bc36010201R278